Test/Unit/Controller/Cart/EmailTest.php

```diff
@@ -0,0 +1,164 @@
+<?php
+
+namespace Bolt\Boltpay\Test\Unit\Controller\Cart;
+
+use Bolt\Boltpay\Controller\Cart\Email;
+use Bolt\Boltpay\Helper\Bugsnag;
+use Bolt\Boltpay\Helper\Cart as CartHelper;
+use Magento\Checkout\Model\Session as CheckoutSession;
+use Magento\Customer\Model\Session as CustomerSession;
+use Magento\Framework\App\Action\Context;
+use Magento\Framework\App\RequestInterface;
+use Magento\Framework\Exception\LocalizedException;
+use Magento\Quote\Model\Quote;
+use PHPUnit\Framework\TestCase;
+
+class EmailTest extends TestCase
+{
+    /**
+     * @var Context
+     */
+    private $context;
+
+    /**
+     * @var CheckoutSession
+     */
+    private $checkoutSession;
+
+    /**
+     * @var CustomerSession
+     */
+    private $customerSession;
+
+    /**
+     * @var Bugsnag
+     */
+    private $bugsnag;
+
+    /**
+     * @var CartHelper
+     */
+    private $cartHelper;
+
+    /**
+     * @var Quote quote
+     */
+    private $quote;
+
+    /**
+     * @var RequestInterface request
+     */
+    private $request;
+
+    /**
+     * @var Email
+     */
+    private $currentMock;
+
+    public function setUp()
+    {
+        $this->initRequiredMocks();
+        $this->initCurrentMock();
+    }
+
+    private function initRequiredMocks()
+    {
+        //objects needed for constructor
+        $this->context = $this->createMock(Context::class);
+        $this->checkoutSession = $this->createMock(CheckoutSession::class);
+        $this->customerSession = $this->createMock(CustomerSession::class);
+
+        $this->bugsnag = $this->createMock(Bugsnag::class);
+
+        $this->cartHelper = $this->createMock(CartHelper::class);
+
+        //objects needed for method
+        $this->quote = $this->createMock(Quote::class);
+        $this->quote->method('getId')->willReturn('1234');
+
+        $this->request = $this->createMock(RequestInterface::class);
+
+        //methods
+        $this->cartHelper->method('quoteResourceSave')->willReturn(null);
+        $this->context->method('getRequest')->willReturn($this->request);
+    }
+
+    private function initCurrentMock()
+    {
+        $this->currentMock = $this->getMockBuilder(Email::class)
+            ->setConstructorArgs([
+                $this->context,
+                $this->checkoutSession,
+                $this->customerSession,
+                $this->bugsnag,
+                $this->cartHelper
+            ])
+            ->enableProxyingToOriginalMethods()
+            ->getMock();
+        return $this->currentMock;
```

Comment:
This line seems redundant, the returned value is never used directly.

Reply:
As with the discussion about mocking the email class, this was just something I followed from `PaymentTest.php`, though I see it's not in your recent commits. Removing.